docs/config.rst

```diff
@@ -0,0 +1,116 @@
+
+Second, there will be some configuration that is going to be shared between many if not all codes.  One example might be database access; if we have a central internal PIT database for coordinating processes and data products, we will need to configure access to that database (host locations, users, passwords, etc.).  By using a shared config system, that only needs to be configured *once*, and then every individual code's config can include the database config, and then we can be sure everybody's using the same thing.
+
+Ultimately, we hope to have a single master config file that configures *all* of the SN PIT codes.  (In practice, this may not truly be a single file, but may include other files, as the config system allows for that.  However, from the point of view of the code that uses it, every code would load the same config file and would have access to all of the config variables.)
+
+
+.. _config-quickstart-example:
+
+Quickstart Example
+==================
+
+At the moment, we do not have any global configs for the SN PIT, so you can create your own config files.  Create a config file ``demo.yaml`` *somewhere* that has contents::
+
+  option1: test1
+  many_options:
+    sub1: subtest1
+    sub2: subtest2
+  three_numbers: [1,2,3]
+
+Next, create a file ``demo.py`` that has contents::
+
+  from snappl.config import Config
+
+  cfg = Config.get()
+
+  print( f"option1 is {cfg.value('option1')}" )
+  print( f"sub option 1 of many options is {cfg.value('many_options.sub1')}" )
+  print( f"three numbers are {cfg.value('three_numbers')}" )
+  print( f"The second number is {cfg.value('three_numbers.1')}" )
+  
+Make sure that you have :ref:`installed snappl <installation-docs>`.  Then run::
+
+  SNAPPL_CONFIG=$PWD/demo.yaml python demo.py
+
+.. _config-quickstart-example-command-line:
+
+Quickstart Example — comand line overriding
+-------------------------------------------
+
+Edit your ``demo.py`` file from the previous example so that it has contents::
+
+  import argparse
+
+  from snappl.config import Config
+
+  cfg = Config.get()
+
+  parser = argparse.ArgumentParser()
+  cfg.augment_argparse( parser )
+  args = parser.parse_args( arglist )
+  cfg.parse_args( args )
+
+  print( f"option1 is {cfg.value('option1')}" )
+  print( f"sub option 1 of many options is {cfg.value('many_options.sub1')}" )
+  print( f"three numbers are {cfg.value('three_numbers')}" )
+  print( f"The second number is {cfg.value('three_numbers.1')}" )
+
+Now try running each of::
+
+  SNAPPL_CONFIG=$PWD/demo.yaml python demo.py
+  SNAPPL_CONFIG=$PWD/demo.yaml python demo.py --option1 hello --three_numbers 4 5 6 7 8
```

Comment:
Is there some pedagogical point that `three_numbers` is a 5-number list here?  My reading is that there is not a check that `three_numbers` only has three elements.